### PR TITLE
fix: custom plugin to fix esbuild esm imports

### PR DESCRIPTION
**Pull request by SiTaggart**
The plugin we used to fix the esbuild issue with require statements in esm compiled packages reintroduced an issue with incorrectly importing defaults from json files, using `import * as x from 'x'`.

This is because it blanketly transformed all import statements, even though the bug we were trying to fix was just require statements.

This customized version of the plugin only modifies imports that are require statements and leaves already correct import statements alone.

### CODE REVIEW

File: tools/build/build-plugins/EsmExternals.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  // Escape characters with special meaning either inside or outside character sets.
+  // Use a simple backslash escape when it’s always valid, and a `\xnn` escape when the simpler form would be disallowed by Unicode patterns’ stricter grammar.
+  return string.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&').replace(/-/g, '\\x2d');
+}
+
+const NAME = 'esm-externals';
+const NAMESPACE = NAME;
+
+// this is a modified version of https://github.com/remorses/esbuild-plugins/tree/master/esm-externals
```
Comment (zahnster) [inline tools/build/build-plugins/EsmExternals.js:16]: praise: nice commenting, much appreciated

File: tools/build/esbuild.js [hunk 2]
```diff
@@ -1,25 +1,6 @@
...
-    });
-
-    // Change all .es6 design-token imports to .common
-    build.onResolve({filter: /\/tokens.es6$/}, ({path}) => {
-      return {path: path.replace('.es6', '.common'), external: true};
-    });
-  },
-};
+const {EsmExternalsPlugin} = require('./build-plugins/EsmExternals');
+const {PasteCJSResolverPlugin} = require('./build-plugins/PasteCJSResolver');
```
Comment (zahnster) [inline tools/build/esbuild.js:3]: praise: nice code organization

**Review by zahnster — APPROVED** [review]
Obama Not Bad meme

**Review by shleewhite — APPROVED** [review]

**Comment by zahnster** [comment]
Great additions, and look at all those green checks! ✅

**Pull request merged**